ground/src/main/java/com/google/android/ground/ui/datacollection/TaskSequenceHandler.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection
+
+import com.google.android.ground.model.submission.TaskData
+import com.google.android.ground.model.task.Task
+
+interface TaskSequenceHandler {
```

Comment:
Do we really expect to have to override this? If not, consider making this concrete and removing the need for a separate Impl class. Alternatively, we could make this an `open` class to bundle the interface with the default implementation (if we do expect to need to override these with alternatives).

Reply:
That's a great point! We won't have another fake implementation of this interface as the normal one can be directly used in unit tests. Will drop the interface and merge with concrete class

Reply:
Done